### PR TITLE
Add conversion from v1beta1 to v1alpha1

### PR DESCRIPTION
**Pull request by pavolloffay**
**Description:** <Describe what has changed.>

The conversion webhook requires converting from old to new version and vice-versa. 

**Link to tracking Issue(s):** <Issue number if applicable>

Updates #2620 

- Resolves: #issue-number

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: apis/v1alpha1/convert.go [hunk 1]
```diff
@@ -175,3 +175,170 @@ func TargetAllocatorEmbedded(in OpenTelemetryTargetAllocator) v1beta1.TargetAllo
 	}
 	return out
 }
+
+func tov1alpha1(in v1beta1.OpenTelemetryCollector) (*OpenTelemetryCollector, error) {
```
Comment (jaronoff97) [inline apis/v1alpha1/convert.go:179]: why do we need the conversion going this direction?

Comment (pavolloffay) [reply apis/v1alpha1/convert.go:179]: It is mentioned in the PR description

File: apis/v1alpha1/convert_test.go [hunk 2]
```diff
@@ -48,9 +47,12 @@ service:
       processors: [resourcedetection]
       exporters: [otlp]
 `
+
+func Test_v1alpha1tov1beta1(t *testing.T) {
+	t.Run("valid config", func(t *testing.T) {
 		cfgV1 := OpenTelemetryCollector{
 			Spec: OpenTelemetryCollectorSpec{
```
Comment (swiatekm-sumo) [inline apis/v1alpha1/convert_test.go:54]: I'd love to see one full smoke test where we set every field in the spec.

Comment (pavolloffay) [reply apis/v1alpha1/convert_test.go:57]: added

**Review by swiatekm-sumo — APPROVED** [review]

**Review by jaronoff97 — APPROVED** [review]

**Pull request merged**